### PR TITLE
Feature/query cxx as

### PR DESCRIPTION
**Pull request by AshleyStanfield-NOAA**
## Description

This PR adds yaml files for SatWind AVHRR NC005080. More specifically, adds the new bufr_ncep_satwind_avhrr.yaml

This PR Adds:
- ./test/testinput/bufr_ncep_satwind_avhrr.yaml
- ./test/testinput/bufr_ncep_satwind_avhrr.bufr_d
- ./test/testoutput/bufr_ncep_satwind_avhrr.nc

and updates: 
- ./test/CMakeLists.txt.

### Issue(s) addressed

#791 - Create Subset 080 - NOAA-15/18/19 YAML for AVHRR IRLW

## Acceptance Criteria (Definition of Done)

- No red flags.
- Data validation checks out. Plot verified.

## Dependencies

None

## Impact

None

### CODE REVIEW

File: test/CMakeLists.txt
Comment (ilianagenkova) [inline test/CMakeLists.txt:221]: Did you add lines 221-222 in an earlier PR? Whoever did, there must be  matching testinput/*.bufr_d and testinput/*.yaml for them. This is not part of your current PR, but I spotted it, so we clean it or fix it.

Comment (ilianagenkova) [reply test/CMakeLists.txt:221]: I looked around, this is a non-problem, it has matching testinput/* files, but I didn't see them at first.

**Review by ilianagenkova — APPROVED** [review]
@AshleyStanfield-NOAA and @emilyhcliu , this looks good!
One reminder - once Ashley add all other NC0050* subtypes, let me know, so I clean up the soon-to-be-obsolete testinput/*satwind_old/new_yaml*  files and the related tests. Thanks!

**Review by rmclaren — COMMENTED** [review]
I usually try to name the the bufr and output files the same as the yaml files so bufr_ncep_satwind_avhrr.bufr_d and bufr_ncep_satwind_avhrr.nc. This way you know exactly which files go together... Does it make sense in this case?

**Comment by AshleyStanfield-NOAA** [comment]
Oh that makes sense!  I've updated the yaml and the files to match now.
modified:   bufr_ncep_satwind_avhrr.yaml
renamed:    testinput/NC005080.bufr_d -> testinput/bufr_ncep_satwind_avhrr.bufr_d
renamed:    testoutput/Out_NC005080.nc -> testoutput/bufr_ncep_satwind_avhrr.nc

**Comment by rmclaren** [comment]
Awesome, thanks!

**Review by rmclaren — APPROVED** [review]

File: test/bufr_ncep_satwind_avhrr.yaml
Comment (emilyhcliu) [inline test/bufr_ncep_satwind_avhrr.yaml:80]: @AshleyStanfield-NOAA The output should be stored in testrun directory, not under your account.

File: test/bufr_ncep_satwind_avhrr.yaml
Comment (emilyhcliu) [inline test/bufr_ncep_satwind_avhrr.yaml:14]: @AshleyStanfield-NOAA The input data should be in ./testinput, not in your own directory.

Comment (AshleyStanfield-NOAA) [reply test/bufr_ncep_satwind_avhrr.yaml]: Thanks Emily, the input and output fields have been updated.

**Comment by emilyhcliu** [comment]
@AshleyStanfield-NOAA I think you checked in your YAML into the wrong directory.
You checked your YAML file (bufr_ncep_satwind_avhrr.yaml)  into ./test directory
It should be checked into ./test/testinput directory.

Did you run ctest of your own yaml?  It should give you an error.

**Comment by emilyhcliu** [comment]
@AshleyStanfield-NOAA I got error messages from the following queries:

```
          heightAssignmentMethod:
            query: "*/HAMD[1]"

          pressure:
            query: "*/PRLC[1]"

          windDirection:
            query: "*/WDIR"

          windSpeed:
            query: "*/WSPD"

          generatingApplication:
            query: "*/GQCPRMS/GNAP[1]"

          percentConfidence:
            query: "*/GQCPRMS/PCCF[1]"
```    

One of the error messages and other threes are similar:
```
PRE-MAIN-ERROR Exception: Bad parameter: Invalid index in query str */PRLC[1]. 
terminate called after throwing an instance of 'eckit::BadParameter'
  what():  Bad parameter: Invalid index in query str */PRLC[1].
```

It looks like you do not need to add [1] anymore.  
@rmclaren Could you confirm if this is true?

**Comment by emilyhcliu** [comment]
@rmclaren @ilianagenkova Please do not merge this one yet.  There are still some issues need to be resolved.  Please see my previous comments above.

**Comment by emilyhcliu** [comment]
@rmclaren @AshleyStanfield-NOAA @ilianagenkova 
I used the full satwind file as an input.
I saw that Ashely is using a satwind AVHRR subset as input.
And the index [1] is required to make the conversion works
```
          heightAssignmentMethod:
            query: "*/HAMD[1]"

          pressure:
            query: "*/PRLC[1]"

          generatingApplication:
            query: "*/GQCPRMS/GNAP[1]"

          percentConfidence:
            query: "*/GQCPRMS/PCCF[1]"
```

File: test/bufr_ncep_satwind_avhrr.yaml
Comment (emilyhcliu) [inline test/bufr_ncep_satwind_avhrr.yaml:44]: Remove these variables.
And, add the following:
```
          timestamp:
            datetime:
              year: "*/YEAR[1]"
              month: "*/MNTH[1]"
              day: "*/DAYS[1]"
              hour: "*/HOUR[1]"
              minute: "*/MINU[1]"
```

File: test/bufr_ncep_satwind_avhrr.yaml
Comment (emilyhcliu) [inline test/bufr_ncep_satwind_avhrr.yaml:131]: Remove these lines and add the following:

```
        - name: "MetaData/dateTime"
          coordinates: "longitude latitude"
          source: variables/timestamp
          longName: "Datetime"
          units: "seconds since 1970-01-01T00:00:00Z"
```

Comment (emilyhcliu) [reply test/bufr_ncep_satwind_avhrr.yaml:131]: The mnemonics YEAR, MNTH, DAYS, HOUR and MINU from BUFR are used to construct EPOCH time (dateTime) in IODA.  The definition is seconds since 1970-01-01T00:00:00Z.

Comment (emilyhcliu) [reply test/bufr_ncep_satwind_avhrr.yaml:131]: @rmclaren @ilianagenkova Let's be a bit more careful in reviewing the PRs related to feature/query_cxx.  There is no hurry.  We have time .  Accuracy is more important than fast and more!

**Comment by ilianagenkova** [comment]
@AshleyStanfield-NOAA , I will begin reviewing the PR

 You could edit this PR (up at the top) and put the new filenames in the text:
"This PR Adds:
/test/bufr_ncep_satwind_avhrr.yaml
/testinput/NC005080.bufr_d
/testoutput/Out_NC005080.nc "

File: test/CMakeLists.txt
Comment (ilianagenkova) [inline test/CMakeLists.txt:147]: @AshleyStanfield-NOAA , please reflect here (in CMakeLists.txt) the new file name(s), i.e.
testinput/NC005080.bufr_d will become testinput/bufr_ncep_satwind_avhrr.bufr_d, etc.

File: test/CMakeLists.txt
Comment (ilianagenkova) [inline test/CMakeLists.txt:223]: Reflect the new filename here, i.e.
testinput/NC005080.nc will become testinput/bufr_ncep_satwind_avhrr.nc

File: test/CMakeLists.txt
Comment (ilianagenkova) [reply test/CMakeLists.txt:221]: @rmclaren , @emilyhcliu , lines 221-222  (and the matching testinput lines and tests) are no longer needed, I'll open a PR to clean this up.

File: test/bufr_ncep_satwind_avhrr.yaml
Comment (ilianagenkova) [reply test/bufr_ncep_satwind_avhrr.yaml:131]: Agree! I'll let @AshleyStanfield-NOAA address the PR suggested changes, and I'll clean up my old lines of code.

**Review by ilianagenkova — CHANGES_REQUESTED** [review]

File: test/bufr_ncep_satwind_avhrr.yaml
Comment (emilyhcliu) [inline test/bufr_ncep_satwind_avhrr.yaml:84]: Remove line 81-84 and add the following:

```
      dimensions:
        - name: Confidence
          paths:
            - "*/GQCPRMS"
```

File: test/bufr_ncep_satwind_avhrr.yaml
Comment (AshleyStanfield-NOAA) [reply test/bufr_ncep_satwind_avhrr.yaml:44]: The yaml has now been changed to match the above. Will be uploaded shortly.

File: test/bufr_ncep_satwind_avhrr.yaml
Comment (AshleyStanfield-NOAA) [reply test/bufr_ncep_satwind_avhrr.yaml:131]: The yaml has now been changed to match the above. Will be uploaded shortly.

File: test/CMakeLists.txt
Comment (AshleyStanfield-NOAA) [reply test/CMakeLists.txt:147]: I've updated CMakeLists to match the above. Will be re-uploaded shortly.

File: test/CMakeLists.txt
Comment (AshleyStanfield-NOAA) [reply test/CMakeLists.txt:223]: I've updated CMakeLists to match the above. Will be re-uploaded shortly.

File: test/bufr_ncep_satwind_avhrr.yaml
Comment (AshleyStanfield-NOAA) [reply test/bufr_ncep_satwind_avhrr.yaml:84]: I've updated the yaml to match the above. Will be re-uploaded shortly.

File: test/bufr_ncep_satwind_avhrr.yaml
Comment (AshleyStanfield-NOAA) [reply test/bufr_ncep_satwind_avhrr.yaml:44]: Actually, I had a question on this edit;  Is "seconds" no longer included?  The update only includes down to minutes.

**Comment by AshleyStanfield-NOAA** [comment]
I've updated the following files.
modified:   CMakeLists.txt
renamed and modified:    bufr_ncep_satwind_avhrr.yaml -> testinput/bufr_ncep_satwind_avhrr.yaml

The input fields and linked name fields for "time" has been updated as suggested.
The CMakeList has been updated to match with the new naming scheme.
The "dimensions" field has been updated as suggested.
And the yaml has now been moved from ./test to ./test/testinput.

I believe that covers the changes mentioned.

File: test/bufr_ncep_satwind_avhrr.yaml
Comment (rmclaren) [reply test/bufr_ncep_satwind_avhrr.yaml]: Seconds are optional. Include if they are available (not all BUFR files have seconds)

File: test/bufr_ncep_satwind_avhrr.yaml
Comment (rmclaren) [reply test/bufr_ncep_satwind_avhrr.yaml]: Sorry Emily. I just didn't see it.

File: test/bufr_ncep_satwind_avhrr.yaml
Comment (rmclaren) [reply test/bufr_ncep_satwind_avhrr.yaml]:
```
          heightAssignmentMethod:
            query: "*/HAMD[1]"

          pressure:
            query: "*/PRLC[1]"

          generatingApplication:
            query: "*/GQCPRMS/GNAP[1]"

          percentConfidence:
            query: "*/GQCPRMS/PCCF[1]"
```

Those indices on HAMD and PRLC should not be needed at all (but should'nt hurt either) investigating.

**Comment by rmclaren** [comment]
So the first instance of the sequence GQCPRMS is associated with wind direction (WDIR) and the second instance as associated with wind speed (WSPD). I think you'll want both of them. So you will need to read in 4 variables from bufr (maybe call them windSpeedGeneratingApp, windDirGeneratingApp, windSpeedPercentConf, windDirPercentConf) and then when you write these out to ioda you will want to use special output "paths" for these 4 parameters. I think the following: `PercentConfidence/windSpeed`, `PercentConfidence/windDirection`, `GeneratingApplication/windSpeed`, and `GeneratingApplication/windDirection`.

File: test/testinput/bufr_ncep_satwind_avhrr.yaml
Comment (rmclaren) [reply test/testinput/bufr_ncep_satwind_avhrr.yaml:152]: Instead of sticking them in `MetaData`.

**Review by rmclaren — CHANGES_REQUESTED** [review]

**Review by rmclaren — CHANGES_REQUESTED** [review]

File: test/testinput/bufr_ncep_satwind_avhrr.yaml
Comment (AshleyStanfield-NOAA) [reply test/testinput/bufr_ncep_satwind_avhrr.yaml]: Indices for HAMD and PRLC have been removed

File: test/testinput/bufr_ncep_satwind_avhrr.yaml
Comment (AshleyStanfield-NOAA) [reply test/testinput/bufr_ncep_satwind_avhrr.yaml:82]: Thank you for letting me know!

File: test/testinput/bufr_ncep_satwind_avhrr.yaml
Comment (AshleyStanfield-NOAA) [reply test/testinput/bufr_ncep_satwind_avhrr.yaml]: I added in two extra pulls/indices to account for wind Speed/Direction GeneratingApplication/PercentConfidence.

File: test/testinput/bufr_ncep_satwind_avhrr.yaml
Comment (AshleyStanfield-NOAA) [reply test/testinput/bufr_ncep_satwind_avhrr.yaml]: I removed the indices and added in seconds.

File: test/testinput/bufr_ncep_satwind_avhrr.yaml
Comment (AshleyStanfield-NOAA) [reply test/testinput/bufr_ncep_satwind_avhrr.yaml]: I like the idea, though how would I go about implementing this? If I used the following set, does the bottom half (name/source/longName/units) only matter in regards to the order in which they are listed?

```
          windDirection:
            query: "*/WDIR"

          windSpeed:
            query: "*/WSPD"

          windSpeed:
            query: "*/GQCPRMS/GNAP[1]"

          windDirection:
            query: "*/GQCPRMS/GNAP[2]"

          windSpeed:
            query: "*/GQCPRMS/PCCF[1]"

          windDirection:
            query: "*/GQCPRMS/PCCF[2]"
```
So then the bottom would just be:
```
        - name: "ObsValue/windDirection"
          source: variables/windDirection
          longName: "Wind Direction"
          units: "deg"

        - name: "ObsValue/windSpeed"
          source: variables/windSpeed
          longName: "Wind Speed"
          units: "m s-1"

        - name: "GeneratingApplication/windSpeed"
          source: variables/windSpeed
          longName: "Wind Speed Generating Application"
          units: ""

        - name: "GeneratingApplication/windDirection"
          source: variables/windDirection
          longName: "Wind Direction Generating Application"
          units: ""

        - name: "PercentConfidence/windSpeed"
          source: variables/windSpeed
          longName: "Wind Speed Per Cent Confidence"
          units: "percent"

        - name: "PercentConfidence/windDirection"
          source: variables/windDirection
          longName: "Wind Direction Per Cent Confidence"
          units: "percent"
```
My assumption is that this wouldn't work given the repeated sources.

Comment (rmclaren) [reply test/testinput/bufr_ncep_satwind_avhrr.yaml]: Your close, but the variable names (in the exports section) must all have unique names. That is to say they must all be named something different (otherwise we can't tell them apart). So instead `*/GQCPRMS/GNAP[1]` being named `windSpeed` it should be called something like `windSpeedGeneratingApp`. The only one that is named `windSpeed` should be the one thats associated with the `*/WSPD` BUFR query. The same with the other variables.

**Comment by emilyhcliu** [comment]
gsi_ncep_satwnd_uwind_poles

**Converted wind speed and direction to u-wind component** 
ioda_ncep_satwnd_uwind_poles
gsi_ncep_satwnd_vwind_poles
**Converted wind speed and direction to v-wind component** 
ioda_ncep_satwnd_vwind_poles

File: test/testinput/bufr_ncep_satwind_avhrr.yaml
Comment (AshleyStanfield-NOAA) [reply test/testinput/bufr_ncep_satwind_avhrr.yaml]: Okay, that was my understanding.

Currently, I have it set as :
          windDirection:
            query: "*/WDIR"

          windSpeed:
            query: "*/WSPD"

          windSpeedGeneratingApplication:
            query: "*/GQCPRMS/GNAP[1]"

          windDirectionGeneratingApplication:
            query: "*/GQCPRMS/GNAP[2]"

          windSpeedPercentConfidence:
            query: "*/GQCPRMS/PCCF[1]"

          windDirectionPercentConfidence:
            query: "*/GQCPRMS/PCCF[2]"

File: test/testinput/bufr_ncep_satwind_avhrr.yaml
Comment (emilyhcliu) [reply test/testinput/bufr_ncep_satwind_avhrr.yaml]: > Will want to read 4 parameters here. `*/GQCPRMS/GNAP[1]`, `*/GQCPRMS/PCCF[1]` for wind direction (WDIR) and `*/GQCPRMS/GNAP[2]`, `*/GQCPRMS/PCCF[2]` for wind speed (WSPD). Give them appropriate names like `windSpeedGeneratingApp`... etc

@rmclaren Nice catch!  I will need to discuss the naming convention for these variables.

File: test/testinput/bufr_ncep_satwind_avhrr.yaml
Comment (emilyhcliu) [reply test/testinput/bufr_ncep_satwind_avhrr.yaml]: Oops! my bad.  Right, no indices are needed for this.

File: test/testinput/bufr_ncep_satwind_avhrr.yaml
Comment (emilyhcliu) [reply test/testinput/bufr_ncep_satwind_avhrr.yaml]: I will discuss the naming convention for these variables with Ryan tomorrow.  They are likely to stay as MetaData group.  
I

File: test/testinput/bufr_ncep_satwind_avhrr.yaml
Comment (rmclaren) [reply test/testinput/bufr_ncep_satwind_avhrr.yaml]: Thats perfect.

File: test/testinput/bufr_ncep_satwind_avhrr.yaml
Comment (emilyhcliu) [inline test/testinput/bufr_ncep_satwind_avhrr.yaml:105]: Change this to "MetaData/windComputationMethod"

(This is from the [JEDI convention spreadsheet](https://docs.google.com/spreadsheets/d/1fHb2QIHOwgp4d6plyn9Dj9HZFZsDRxcPaUV6hdTb5BU/edit#gid=0))

File: test/testinput/bufr_ncep_satwind_avhrr.yaml
Comment (emilyhcliu) [inline test/testinput/bufr_ncep_satwind_avhrr.yaml:115]: Change to "MetaData/dataProviderOrigin"

File: test/testinput/bufr_ncep_satwind_avhrr.yaml
Comment (emilyhcliu) [inline test/testinput/bufr_ncep_satwind_avhrr.yaml:120]: Change to MetaData/sensorCentralFrequency

File: test/testinput/bufr_ncep_satwind_avhrr.yaml
Comment (emilyhcliu) [inline test/testinput/bufr_ncep_satwind_avhrr.yaml:125]: Change to "MetaData/qualityFlags"

File: test/testinput/bufr_ncep_satwind_avhrr.yaml
Comment (emilyhcliu) [inline test/testinput/bufr_ncep_satwind_avhrr.yaml:130]: Change MetaData/windHeightAssignMethod

**Comment by emilyhcliu** [comment]
Suggestion: Check the following two documents while making decision on the variable names:

- [JEDI naming convention](https://docs.google.com/spreadsheets/d/1fHb2QIHOwgp4d6plyn9Dj9HZFZsDRxcPaUV6hdTb5BU/edit#gid=0)
- [ObsSpace.yaml](https://github.com/JCSDA-internal/ioda/blob/feature/query_cxx/share/ioda/yaml/validation/ObsSpace.yaml)

If you can not find a proper name from the above, check the following based on the BUFR mnemonics (FXXYYY)

- [WMO Element Table](https://confluence-test.ecmwf.int/display/ECC/WMO%3D35+element+table)

File: test/testinput/bufr_ncep_satwind_avhrr.yaml
Comment (AshleyStanfield-NOAA) [reply test/testinput/bufr_ncep_satwind_avhrr.yaml]: The yaml has been updated.
```
          windComputationMethod:
            query: "*/SWCM"

        - name: "MetaData/windComputationMethod"
          source: variables/windComputationMethod
          longName: "Satellite Derived Wind Computation Method"
          units: ""
```

File: test/testinput/bufr_ncep_satwind_avhrr.yaml
Comment (AshleyStanfield-NOAA) [reply test/testinput/bufr_ncep_satwind_avhrr.yaml]: The yaml has been updated.
```
          dataProviderOrigin:
            query: "*/OGCE"
            
        - name: "MetaData/dataProviderOrigin"
          source: variables/dataProviderOrigin
          longName: "Identification of Originating/Generating Center"
          units: ""
```

File: test/testinput/bufr_ncep_satwind_avhrr.yaml
Comment (AshleyStanfield-NOAA) [reply test/testinput/bufr_ncep_satwind_avhrr.yaml]: The yaml has been updated.
```
          sensorCentralFrequency:
            query: "*/SCCF"

        - name: "MetaData/sensorCentralFrequency"
          source: variables/sensorCentralFrequency
          longName: "Satellite Channel Center Frequency"
          units: "Hz"
```

File: test/testinput/bufr_ncep_satwind_avhrr.yaml
Comment (AshleyStanfield-NOAA) [reply test/testinput/bufr_ncep_satwind_avhrr.yaml]: The yaml has been updated:
```
          qualityFlags:
            query: "*/SWQM"
 
         - name: "MetaData/qualityFlags"
          source: variables/qualityFlags
          longName: "SDMEDIT Satellite Wind Quality Mark"
          units: ""
```

File: test/testinput/bufr_ncep_satwind_avhrr.yaml
Comment (AshleyStanfield-NOAA) [reply test/testinput/bufr_ncep_satwind_avhrr.yaml]: The yaml has been updated:
```
          windHeightAssignMethod:
            query: "*/HAMD"
            
        - name: "MetaData/windHeightAssignMethod"
          source: variables/windHeightAssignMethod
          longName: "Wind Height Assignment Method"
          units: ""
```

**Review by rmclaren — COMMENTED** [review]
Your changes look good to me, but I'll leave it to Emily to decide when we are ready to merge the changes. :) Don't forget to address the issues with the generatingApplication and percentConfidence variables (naming of ioda output as noted).

File: test/testinput/bufr_ncep_satwind_avhrr.yaml
Comment (AshleyStanfield-NOAA) [reply test/testinput/bufr_ncep_satwind_avhrr.yaml:150]: I'm admittedly still a bit confused from before, as if I change that variable over to windSpeed, I end up with the following:

```
          windSpeed:
            query: "*/WSPD"

          windSpeed:
            query: "*/GQCPRMS/GNAP[1]"
            
-------------------------------------

        - name: "ObsValue/windSpeed"
          source: variables/windSpeed
          longName: "Wind Speed"
          units: "m s-1"

        - name: "GeneratingApplication/windSpeed"
          source: variables/windSpeed
          longName: "Wind Speed Generating Application"
          units: ""
```

Will the yaml function with both variables being named "windSpeed"?

Comment (emilyhcliu) [reply test/testinput/bufr_ncep_satwind_avhrr.yaml:150]: @rmclaren I spoke with JCSDA (Ryan, Greg, ...etc).   The GeneratingApplication and PercentConfident should stay with MetaData group.

**Comment by emilyhcliu** [comment]
@rmclaren  and @AshleyStanfield-NOAA I spoke with JCSDA (Ryan, Greg, ...etc).   The GeneratingApplication and PercentConfident should stay with MetaData group.  Let's use the following:

MetaData/windSpeedPercentConfidence
MetaData/windDirectionPercentConfidence

MetaData/windSpeedGeneratingApplication
MetaData/windDirectionGeneratingApplication

File: test/testinput/bufr_ncep_satwind_avhrr.yaml
Comment (emilyhcliu) [reply test/testinput/bufr_ncep_satwind_avhrr.yaml:14]: Good catch.

Comment (AshleyStanfield-NOAA) [reply test/testinput/bufr_ncep_satwind_avhrr.yaml:14]: Absolutely. I'll remove it.

**Review by emilyhcliu — APPROVED** [review]

**Comment by rmclaren** [comment]
@emilyhcliu Thats good to know. Thanks for looking into it.

**Comment by rmclaren** [comment]
@ilianagenkova Do you think its OK to merge this one, or do you still have comments that you think need to be addressed?

**Comment by AshleyStanfield-NOAA** [comment]
Just wanted to check in as this may be the best place for it, but is there any standard procedure for how you would like me to go about setting up the next pull request?

More specifically, I followed Praveen's footsteps in this pull request for Feature/query_cxx_as which covers avhrr, but I can finalize the yamls for goes-r, himawari, and meteosat in short order, and I'm just curious if we have a best approach for this.

I believe that the current yaml could essentially be incorporated to include all files, if we are able to modify it in such a way to read one of two variables, depending on which one is available:
```
  windHeightAssignMethod:
            query: "*/HAMD"
            query: "*/EHAM"
            
            longName: "Wind Height Assignment Method"
            vs
            longName: "Wind Extended Height Assignment Method"
```
and other steps along this nature... or are we set on creating a yaml for each one individually?
If we're generating a new yaml for each set, shall I just make essentially pull and branch feature_query_css_AS_GOES and go through this process again using the knowledge gained here?  Happy to do whatever you all feel is best as I'm newer to GitHub.

**Comment by rmclaren** [comment]
@AshleyStanfield-NOAA  You can do that as follows.

```
  windHeightAssignMethod:
            query: "[*/HAMD, */EHAM]"
```

This will use HAMD for subsets that have HAND available otherwise EHAM.


I think the long name could just be.
```        
            longName: "Wind Height Assignment Method"
```

If the long name is required to be different you might be better off reading two separate variables and treating them separately. In that case you'll just get missing values (10e10) for these values on subsets where they don't exist.

Is this solution what you were for?

**Comment by AshleyStanfield-NOAA** [comment]
Thanks @rmclaren !  That, may be exactly what I'm looking for.  I'd like to revisit this again with @ilianagenkova, and Iliana, we can decide on a plan. After deciding if we want to generate multiple yamls, or to try for a single all-purpose yaml, I can probably generate the other 3 in very very short order. Realistically, there are minimal changes between each set, just a few different variable names and long-names.

**Pull request merged**
